pkg/core/native/designate.go

```diff
@@ -386,14 +386,16 @@ func (s *Designate) DesignateAsRole(ic *interop.Context, r noderoles.Role, pubs
 		return fmt.Errorf("failed to update Designation role data cache: %w", err)
 	}
 
+	itm := stackitem.NewArray([]stackitem.Item{
+		stackitem.NewBigInteger(big.NewInt(int64(r))),
+		stackitem.NewBigInteger(big.NewInt(int64(ic.Block.Index))),
+	})
 	ic.Notifications = append(ic.Notifications, state.NotificationEvent{
 		ScriptHash: s.Hash,
 		Name:       DesignationEventName,
-		Item: stackitem.NewArray([]stackitem.Item{
-			stackitem.NewBigInteger(big.NewInt(int64(r))),
-			stackitem.NewBigInteger(big.NewInt(int64(ic.Block.Index))),
-		}),
+		Item:       itm,
 	})
+	ic.VM.AddRef(itm)
```

Comment:
Maybe we better add some `ic.AddNotification(hash, name, item)` that will handle this refcounting internally (as well as `state.NotificationEvent` creation)?

Reply:
Yes, fixed.